### PR TITLE
More refactoring and tests.

### PR DESCRIPTION
**Pull request by homeworkprod**
As promised :)

### CODE REVIEW

File: gitfiti.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def load_images(img_names):
         return loaded_imgs
 
 
-def get_calendar(username, base_url):
+def retrieve_contributions_calendar(username, base_url):
     """retrieves the GitHub commit calendar data for a username"""
     base_url = base_url + 'users/' + username
```
Comment (gelstudios) [inline gitfiti.py]: this seems wrong to me now, does base_url have a trailing slash? and will it always even if user entered?

Comment (gelstudios) [inline gitfiti.py]: edit: nope 

```
>>> There was a problem fetching data from https://github-server.examplecompany.lanusers/gelstudios/contributions
<urlopen error [Errno 8] nodename nor servname provided, or not known>
```

File: gitfiti.py [hunk 2]
```diff
@@ -219,24 +219,23 @@ def get_calendar(username, base_url):
...
 
-    output = list(output)
-    output.sort()
-    output.reverse()
-    return output[0]
+
+def find_max_daily_commits(contributions_calendar):
+    """finds the highest number of commits in one day"""
+    daily_counts = parse_contributions_calendar(contributions_calendar)
+    return max(daily_counts)
```
Comment (gelstudios) [inline gitfiti.py]: Elegant use of max() over a generator 👍

File: gitfiti.py (hunk 1 above)
Comment (homeworkprod) [inline gitfiti.py]: That should be handled, right. But it's outside of this PR's scope (and isn't touched by it).

Comment (homeworkprod) [inline gitfiti.py]: Also, if the second argument is about allowing to use this with other providers besides GitHub, the URL path should probably be configurable as well (even if, say, GitLab, might provide data under the same path pattern).

**Comment by homeworkprod** [comment]
Thanks, great working with you.

**Pull request merged**